File: Remove Duplicates from Sorted List/Remove Duplicates from Sorted List II.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode temp=head;
        while(temp!=null&&temp.next!=null){
           ListNode future=temp;
           while(future.next!=null){
               if(temp.val==future.next.val){
                   future.next=future.next.next;
               }else{
                   future=future.next;
               }
           }
           temp=temp.next;
        }
        return head;
    }
}

File: Remove Duplicates from Sorted List II/Remove Duplicates from Sorted List II.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class newNode:
    def _init_(self):
        self.next=None
class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
            previous=newNode()
            previous.next=head
            newhead=previous
            temp=newhead
            while temp.next!=None:
                future=temp.next
                while future.next!=None:
                    if future.next.val==future.val:
                        future=future.next
                    else:
                        break
                if future!=temp.next:
                    temp.next=future.next
                else:
                    temp=temp.next
            return newhead.next
